src/app/_components/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+
+interface TextInputProps {
+  id: string;
+  name: string;
+  placeholder?: string;
+  value?: string;
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  onFocus?: () => void;
+  onBlur?: () => void;
+  errorMessage?: string;
+  successMessage?: string;
+}
+
+const TextInput: React.FC<TextInputProps> = ({
+  id,
+  name,
+  placeholder,
+  value,
+  onChange,
+  onFocus,
+  onBlur,
+  errorMessage,
+  successMessage,
+}) => {
+  const isError = Boolean(errorMessage);
```

Comment:
I don't think you need to make these isError or isSuccess variables. You can just use the messages directly to see if it is defined or not.